### PR TITLE
Feature: Share the game link with Opponent

### PR DESCRIPTION
**Pull request by VK-RED**
### What does this PR do ?
- Modified the JOIN_ROOM logic in GameManager.ts to handle the secondPlayer when joined via the game link.
- Created ShareGame Component to display and copy the gameLink.

### Screencast
[Screencast from 06-05-24 10:18:34 PM IST.webm](https://github.com/code100x/chess/assets/130341088/832f0fac-8748-4212-b6e4-04f2d74f7907)

### CODE REVIEW

**Comment by VK-RED** [comment]
Thanks Kirat !! Resolving the conflicts..

**Comment by siinghd** [comment]
@VK-RED will check it soon

**Comment by VK-RED** [comment]
Thanks @siinghd !!

File: apps/frontend/src/screens/Game.tsx [hunk 1]
```diff
@@ -67,7 +68,7 @@ export const Game = () => {
   >(null);
   const [player1TimeConsumed, setPlayer1TimeConsumed] = useState(0);
   const [player2TimeConsumed, setPlayer2TimeConsumed] = useState(0);
-
+  const [gameID,setGameID] = useState("");
```
Comment (siinghd) [inline apps/frontend/src/screens/Game.tsx:71]: why this?

doesnt `const { gameId } = useParams();` work?

File: apps/frontend/src/components/ShareGame.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { useState } from "react"
+import { Button } from "./Button";
+
+export const ShareGame = ({className,gameId}:{className?:string,gameId:string}) => {
+
+    const [url] = useState(window.origin+"/game/"+gameId);
```
Comment (siinghd) [inline apps/frontend/src/components/ShareGame.tsx:6]: why usestate here sir?

File: apps/frontend/src/screens/Game.tsx (hunk 1 above)
Comment (VK-RED) [reply apps/frontend/src/screens/Game.tsx:71]: No ser, that won't work. When the first player initially joins he will be at the route /game/random. After receiving the game_added message we will actually get the game ID, using which we create the link ..

File: apps/frontend/src/components/ShareGame.tsx (hunk 2 above)
Comment (VK-RED) [reply apps/frontend/src/components/ShareGame.tsx:6]: Agreed , will convert to a normal var.

File: apps/frontend/src/screens/Game.tsx (hunk 1 above)
Comment (VK-RED) [reply apps/frontend/src/screens/Game.tsx:71]: Agreed , will convert to a normal var.

File: apps/frontend/src/components/ShareGame.tsx (hunk 2 above)
Comment (VK-RED) [reply apps/frontend/src/components/ShareGame.tsx:6]: Agreed , will convert to a normal var.

File: apps/frontend/src/screens/Game.tsx (hunk 1 above)
Comment (VK-RED) [reply apps/frontend/src/screens/Game.tsx:71]: No ser, that won't work. When the first player initially joins he will be at the route /game/random. After receiving the game_added message we will actually get the game ID, using which we create the link ..

**Comment by VK-RED** [comment]
@siinghd you can check it now ...

File: apps/frontend/src/screens/Game.tsx (hunk 1 above)
Comment (siinghd) [reply apps/frontend/src/screens/Game.tsx:71]: oh, i see the login is different, would be better if we push the user who is wainting to the actual url with the ID: /game/id... 

for now mergin this.

**Pull request merged**